pkg/config/mesh/networks_watcher.go

```diff
@@ -115,19 +128,44 @@ func (w *internalNetworkWatcher) SetNetworks(meshNetworks *meshconfig.MeshNetwor
 		// Store the new config.
 		w.prevNetworks = w.networks
 		w.networks = meshNetworks
-		handlers = append([]func(){}, w.handlers...)
+		handlers = append([]*watchHandler{}, w.handlers...)
 	}
 	w.mutex.Unlock()
 
 	// Notify the handlers of the change.
 	for _, h := range handlers {
-		h()
+		h.handler()
 	}
 }
 
 // AddNetworksHandler registers a callback handler for changes to the mesh network config.
-func (w *internalNetworkWatcher) AddNetworksHandler(h func()) {
+func (w *internalNetworkWatcher) AddNetworksHandler(h func()) WatcherHandlerRegistration {
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+
+	handler := &watchHandler{
+		handler: h,
+	}
+	w.handlers = append(w.handlers, handler)
+	return handler
+}
+
+// DeleteNetworksHandler deregister a callback handler for changes to the mesh network config.
+func (w *internalNetworkWatcher) DeleteNetworksHandler(registration WatcherHandlerRegistration) {
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
-	w.handlers = append(w.handlers, h)
+
+	if len(w.handlers) == 0 {
+		return
+	}
+
+	slices.FilterInPlace(w.handlers, func(handler *watchHandler) bool {
```

Comment:
This seems to be a bug. The return value of `slices.FilterInPlace` must be reassigned to `w.handlers`

Reply:
Again, it seems that the following assertion can be moved outside.